server/routes/users.js

```diff
@@ -0,0 +1,20 @@
+const express = require('express');
+const router = express.Router();
+const user = require('../controller/user');
+
+/* GET users listing. */
+router.get('/', (req, res, next) => {
+    res.send('respond with a resource');
+});
+
+router.post('/', user.create);
+
+router.get('/user/:id', user.find);
+
+router.get('/user', user.findAll);
+
+router.put('/updatebyid', user.updateById);
+
+router.put('/delete', user.deleteById);
```

Comment:
The homework does not specify much how the routes should look like, so this comment would be an extra, but I highly recommend looking into it:

Those routes look a bit off, except the post, that one is correct.

Remember that they are under a prefix `/api/users`. So the getAll route has an unnecessary `user` like: `/api/users/user`

The find user also has an unnecessary `user` like: `/api/users/user/:id`

And you update route should not have `updatebyid`, it should only contain the resouce id i the url like: `PUT /api/users/:id`

Last but not least your Delete route should also have the id on the user and it should use the delete method like `DELETE /api/users/:id`

It should look like this
```javascript
router.get('/', user.findAll);

router.post('/', user.create);

router.get('/:id', user.find);

router.put('/:id', user.updateById);

router.delete('/:id', user.deleteById);
``` 

Some reading on the topic:

In special, read section `Nouns are good; verbs are bad`
https://pages.apigee.com/rs/apigee/images/api-design-ebook-2012-03.pdf

And the very heavy reading, if you feel like it :)
https://www.ietf.org/rfc/rfc2616.txt
In special sections 9.6 and 9.7 for PUT and DELETE.

Reply:
I changed a little bit. But I think the delete function should be look like a UPDATE not a DELETE because it's soft delete. Only change isDeleted property to true.